nucliadb/nucliadb/ingest/tests/unit/consumer/test_utils.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import asyncio
+
+import pytest
+
+from nucliadb.ingest.consumer import utils
+
+pytestmark = pytest.mark.asyncio
+
+
+async def test_delay_task_handler():
+    dth = utils.DelayedTaskHandler(0.05)
+    await dth.initialize()
+
+    counter = 0
+
+    async def handler():
+        nonlocal counter
+        counter += 1
+
+    dth.schedule("key1", handler)
+    dth.schedule("key1", handler)
+    dth.schedule("key1", handler)
```

Comment:
Can you write a test that makes it explicit which of these schedule operations will be ignored?

We want the latter to be effective, don't we?

Reply:
I think I have a better test to show how the utility is meant to work!